examples/ios/swift/Migration/AppDelegate.swift

```diff
@@ -59,85 +26,23 @@ extension UIApplication {
-                    if oldSchemaVersion < 1 {
-                        // combine name fields into a single field
-                        let firstName = oldObject!["firstName"] as! String
-                        let lastName = oldObject!["lastName"] as! String
-                        newObject?["fullName"] = "\(firstName) \(lastName)"
-                    }
-                }
-            }
-            if oldSchemaVersion < 2 {
-                migration.enumerateObjects(ofType: Person.className()) { _, newObject in
-                    // give JP a dog
-                    if newObject?["fullName"] as? String == "JP McDonald" {
-                        let jpsDog = migration.create(Pet.className(), value: ["Jimbo", "dog"])
-                        let dogs = newObject?["pets"] as? List<MigrationObject>
-                        dogs?.append(jpsDog)
-                    }
-                }
-            }
-            print("Migration complete.")
-        }
-
-        Realm.Configuration.defaultConfiguration = Realm.Configuration(schemaVersion: 3, migrationBlock: migrationBlock)
-
-        // print out all migrated objects in the default realm
-        // migration is performed implicitly on Realm access
-        print("Migrated objects in the default Realm: \(try! Realm().objects(Person.self))")
-
-        //
-        // Migrate a realms at a custom paths
-        //
-        if let v1URL = bundleURL("default-v1"), let v2URL = bundleURL("default-v2") {
-            let realmv1URL = defaultParentURL.appendingPathComponent("default-v1.realm")
-            let realmv2URL = defaultParentURL.appendingPathComponent("default-v2.realm")
-
-            let realmv1Configuration = Realm.Configuration(fileURL: realmv1URL, schemaVersion: 2, migrationBlock: migrationBlock)
-            let realmv2Configuration = Realm.Configuration(fileURL: realmv2URL, schemaVersion: 3, migrationBlock: migrationBlock)
-
-            do {
-                try FileManager.default.removeItem(at: realmv1URL)
-                try FileManager.default.copyItem(at: v1URL, to: realmv1URL)
-                try FileManager.default.removeItem(at: realmv2URL)
-                try FileManager.default.copyItem(at: v2URL, to: realmv2URL)
-            } catch {}
-
-            // migrate realms at realmv1Path manually, realmv2Path is migrated automatically on access
-            try! Realm.performMigration(for: realmv1Configuration)
-
-            // print out all migrated objects in the migrated realms
-            let realmv1 = try! Realm(configuration: realmv1Configuration)
-            print("Migrated objects in the Realm migrated from v1: \(realmv1.objects(Person.self))")
-            let realmv2 = try! Realm(configuration: realmv2Configuration)
-            print("Migrated objects in the Realm migrated from v2: \(realmv2.objects(Person.self))")
+        
+        // Choose to either create example files or execute them.
+        let shouldCreateExamples = false
```

Comment:
I thought about the target idea too but that would create a lot of new targets over time when we add more versions. I do think it would be easier for the user though, yes. `xcconfig` makes it more clean on the other side. Mentioning that in the README might be enough.

@jsflax Will we keep the config idea then?

Reply:
Not sure multiple targets make the user's experience simpler though. Config flags make it pretty clear what's going on and easy enough to test

Reply:
Yeah, I think a config file with comments around the guards explaining the "what's and why's" is the cleanest solution here

Reply:
@leemaguire addresses a valid point there though. API changes and compile errors won't get caught. Neither by my solution nor by the pre-compiler flags. Having multiple targets (all included in the `Migration` schema) would solve this quite nicely. @jsflax 

In both cases we would have to explain this in the `README` or some other appropriate location.

Reply:
The examples compilation should be tested on the CI, in which case we would pass the flags to the compiler. Being able to compile this locally isn't a great test to see if something has broken.

But yes, the explanation and instructions need to be in the README.